bin/flutter

```diff
@@ -15,6 +15,11 @@ SCRIPT_PATH="$FLUTTER_TOOLS_DIR/bin/flutter_tools.dart"
 # TODO(abarth): We shouldn't require dart to be on the user's path.
 DART=dart
 
+if [ -n $FLUTTER_DEV ]; then
+  $DART --checked --package-root="$FLUTTER_TOOLS_DIR/packages" "$SCRIPT_PATH" "$@"
+  exit 0
```

Comment:
This skips the `pub get` steps, but I guess we're assuming that dev mode people will know to do those.

Also, you're losing the exit code.  Maybe `exit $?` ?